sigar_linux.go

```diff
@@ -339,6 +351,29 @@ func (self *ProcExe) Get(pid int) error {
 	return nil
 }
 
+func (self *ProcFDUsage) Get(pid int) error {
+	err := readFile(procFileName(pid, "limits"), func(line string) bool {
+		if strings.HasPrefix(line, "Max open files") {
+			fields := strings.Fields(line)
+			if len(fields) == 6 {
+				self.SoftLimit, _ = strconv.ParseUint(fields[3], 10, 64)
+				self.HardLimit, _ = strconv.ParseUint(fields[4], 10, 64)
+				return false
```

Comment:
I think this `return false` should be moved outside and after this `if` block. After finding `Max open files` you want to stop reading the file regardless of success/failure in parsing this line.